src/containers/record-modal.jsx

```diff
@@ -80,17 +80,19 @@ class RecordModal extends React.Component {
                     sampleCount: clippedSamples.length
                 };
 
-                // Load the encoded .wav into the storage cache and get resulting
-                // md5 from storage
-                const storage = this.props.vm.runtime.storage;
-                const md5 = storage.builtinHelper.cache(
+                // Create an asset from the encoded .wav and get resulting md5
+                const storage = this.props.vm.storage;
+                vmSound.asset = storage.createAsset(
```

Comment:
Instead of keeping the data for an asset in storage, keyed by asset id, we just keep an asset object on the vm objects that use the assets.